src/WMOWithRequest/WMOWithRequest.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from 'react';
+import DefaultLoading from './DefaultLoading';
+import DefaultRetryError from './DefaultRetryError';
+
+export const WMORequestResult = {
+  CLIENT_ERROR: 'CLIENT_ERROR',
+  SERVER_ERROR: 'SERVER_ERROR',
+  NETWORK_ERROR: 'NETWORK_ERROR',
+  SUCCESS: 'SUCCESS'
+};
+
+function withRequestWrapper(
+  WrappedComponent,
+  { onRequest, loadingComponent, mapProps, networkErrorComponent, serverErrorComponent }
```

Comment:
mapProps is not very clear. I'd rename it to mapPropsTo<something>.
Any ideas for something?